src/hdf5_back_gen.py

```diff
@@ -0,0 +1,496 @@
+#!/usr/bin/env python
+"""This module generates HDF5 backend code found in src/hdf5_back.cc"""
+import json
+import textwrap
+
+with open('../share/dbtypes.json') as f:
+    RAW_TABLE = json.load(f)
+
+V3_TABLE = set(tuple(row) for row in RAW_TABLE[897:])
+
+def list_dep(orig):
+    """Returns a list of dependent dbtypes.
+    
+    Keyword arguments: 
+    orig -- original dbtype.
+    
+    First element of returned list will always be orig itself.
+    """
+    pieces = orig.split("_")
+    if len(pieces) == 1:
+        return pieces
+    containers1 = ["VECTOR", "SET", "LIST", "VL_VECTOR", "VL_SET", "VL_LIST"]
+    containers2 = ["MAP", "PAIR", "VL_MAP", "VL_PAIR"]
+
+    if pieces[0] == "VL":
+        del pieces[0:2]
+    else:
+        del pieces[0]
+
+    for i in range(0, len(pieces)):
+        if pieces[i] == "VL":
+            pieces[i] = pieces[i] + "_" + pieces[i + 1]
+            del pieces[i + 1]
+        if i == len(pieces) - 1:
+            break
+    
+    for i in range(len(pieces) - 1, -1, -1):
+        if pieces[i] in containers1:
+            pieces[i] += "_" + pieces[i + 1]
+            del pieces[i + 1]
+        elif pieces[i] in containers2:
+            pieces[i]+= "_" + pieces[i + 1] + "_" + pieces[i + 2]
+            del pieces[i + 1:i + 3]
+    pieces = [orig] + pieces
+    return pieces
```

Comment:
My concern with this function is that it relies on the dbtype *name* to infer things about the C++ type *structure*.  As a silly foil, what if there was a "volume-length units" primitive type that happened to be called "VL_UNITS". If I am reading this right, this function would interpret this as variable length. The "VL_" should be interpreted as a convention only, and not a strict rule.  

I think it is much better to rely on the C++ type to find the sub C++ types and lookup all the potential dbtypes that share the sub-C++type.  You might loose the VL information only doing this, but I am not sure that is an issue yet.